activesupport/lib/active_support/testing/assertions.rb

```diff
@@ -5,6 +5,50 @@ module Testing
     module Assertions
       UNTRACKED = Object.new # :nodoc:
 
+      # Asserts that a message matching <tt>regex</tt> was logged
+      # within the provided block. Passes if content logged in
+      # the block matches the regex provided.
+      #
+      #   assert_logs /an informational log message/ do
+      #     Rails.logger.info 'an informational log message'
+      #   end
+      def assert_logs(regex)
+        old_logger = Rails.logger
+        log = StringIO.new
+        Rails.logger = Logger.new(log)
+
+        begin
+          yield
+
+          log.rewind
+          assert_match message, log.read, message
```

Comment:
I think you intend to pass `regex` as the first argument here (and perhaps accept an optional `message` parameter above).